### PR TITLE
feat: bump ESLint, NodeJS, and TS minimum version requirements

### PR DESCRIPTION
**Pull request by bradzacher**
BREAKING CHANGE:
Bumps the minimum dependency requirements

## Overview

Fixes https://github.com/typescript-eslint/typescript-eslint/issues/8346

Breaking changes:
- Bump the minimum NodeJS version to v18.18.0 - which aligns with ESLint v9's minimum.
- Drop support for NodeJS 19 (same as ESLint v9)
- Bump the minimum TypeScript version to v4.7.5
- Bump the minimum ESLint version to v8.56.0
- Remove backwards-compatability helpers from `@typescript-eslint/utils/eslint-utils` (`getAncestors`, `getCwd`, `getDeclaredVariables`, `getFilename`, `getScope`, `getSourceCode`)

Other changes:
- Update ESLint types to match the new minimum version.
- Update the playground so it doesn't allow users to select an unsupported TS version.
- Update the docs so that dependency versions are pulled in automatically.

### CODE REVIEW

File: packages/eslint-plugin/tests/rules/consistent-type-imports.test.ts
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/tests/rules/consistent-type-imports.test.ts:1]: [Testing] I did a full text search for `dependencyConstraints: {` in `eslint-plugin` test files and got 48 results in 9 files:

* `typescript: '3.8',`: 1 result, 1 file
* `typescript: '4.1',`: 35 results, 6 files
* `typescript: '4.3',`: 10 results, 2 files
* `typescript: '4.7',`: 2 results, 1 file

More removals to be done? 🔪

File: packages/typescript-estree/src/parseSettings/warnAboutTSVersion.ts [hunk 1]
```diff
@@ -2,11 +2,11 @@ import semver from 'semver';
 import * as ts from 'typescript';
 
 import type { ParseSettings } from './index';
+
 /**
- * This needs to be kept in sync with /docs/users/Versioning.mdx
- * in the typescript-eslint monorepo
+ * This needs to be kept in sync with package.json in the typescript-eslint monorepo
```
Comment (JoshuaKGoldberg) [inline packages/typescript-estree/src/parseSettings/warnAboutTSVersion.ts:7]: [Praise] Heh, good fix.

File: packages/website/src/components/OptionsSelector.tsx [hunk 2]
```diff
@@ -24,6 +25,8 @@ export interface OptionsSelectorParams {
   readonly tsVersions: readonly string[];
 }
 
+const MIN_TS_VERSION_SEMVER = '>=4.7.5';
```
Comment (JoshuaKGoldberg) [inline packages/website/src/components/OptionsSelector.tsx:28]: [Bug] 4.7 isn't being given as an option in the playground, was this meant to be 4.7.4?

Playground TypeScript version selector showing minors from 4.8 up through 5.3

```suggestion
const MIN_TS_VERSION_SEMVER = '>=4.7.4';
```

**Review by JoshuaKGoldberg — DISMISSED** [review]
LGTM! I trust you on the playground selector 🙂 . This is great - really excited to lose the burden of older ESLint support!

**Comment by JoshuaKGoldberg** [comment]
I ran a few full text regex searches in `.ts` files and found some older ESLint compat support:

* `/eslint.*v7/`
* `/eslint.*v8/

Not blocking IMO, but at least something to note as a good followup.

**Comment by JamesHenry** [comment]
🎉

File: package.json [hunk 3]
```diff
@@ -63,7 +63,6 @@
     "@nx/eslint": "17.2.8",
     "@nx/jest": "17.2.8",
     "@nx/workspace": "17.2.8",
-    "@prettier/sync": "^0.5.0",
```
Comment (JoshuaKGoldberg) [inline package.json:66]: _heck yes_

Comment (JoshuaKGoldberg) [inline package.json:66]: _heck yes_ async everywhere

Comment (bradzacher) [reply package.json:66]: everywhere *except* `eslint-plugin-internal`. One day we'll have async plugins . . . .

**Review by JoshuaKGoldberg — APPROVED** [review]
A girl dressed in victorian clothing (in the Muppet Christmas Carol movie) happily saying 'Wonderful'

File: package.json (hunk 3 above)
Comment (JoshuaKGoldberg) [reply package.json:66]: 🙏

**Pull request merged**